src/canvas.class.js

```diff
@@ -492,6 +492,17 @@
      * @return {Boolean}
      */
     isTargetTransparent: function (target, x, y) {
+      if (target.shouldCache() && target._cacheCanvas && !target.isCacheDirty()) {
+        var normalizedPointer = this._normalizePointer(target, {x: x, y: y}),
```

Comment:
Changed to use _normalizePointer() here, which uses the xform matrix so it handles rotation etc.